solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/EuristicBoostedLambdaMARTModel.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.solr.ltr.ranking;
+
+import java.util.List;
+import org.apache.solr.ltr.util.ModelException;
+import org.apache.solr.ltr.util.NamedParams;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public class EuristicBoostedLambdaMARTModel extends EuristicBoostedModel {
+
+  public EuristicBoostedLambdaMARTModel(String name, List<Feature> features,
+                                        String featureStoreName, List<Feature> allFeatures,
+                                        NamedParams params) throws ModelException {
+    super(name, features, featureStoreName, allFeatures, params);
+    super.LTRModel = new LambdaMARTModel(name, features, featureStoreName, allFeatures, params);
```

Comment:
Perhaps instead of directly setting LTRModel in the constructor we should have an abstract function called getLTRModel inside of EuristicBoostedModel (which should really be HerusticBoostModel :) ) and then the parent constructor should have LTRModel = getLTRModel(name, features, featureStoreName, allFeatures, params) . The reason is that if someone does not set LTRModel it will create errors, but that is not immediately apparent. If the only thing a person needs to do is overwrite an abstract function called getLTRModel it makes it easy for them to understand exactly what they need to do.